services/web/server/src/simcore_service_webserver/scicrunch/_rest.py

```diff
@@ -0,0 +1,122 @@
+"""
+   Functions and models to query scicrunch service REST API (https://scicrunch.org/api/)
+
+    - http client for API requests
+    - Error handling:
+        - translates network errors
+        - translates request error codes
+
+    Free functions with raw request scicrunch.org API
+    - client request context
+    - raise_for_status=True -> Raise an aiohttp.ClientResponseError if the response status is 400 or higher
+    - validates response and prunes using pydantic models
+
+    SEE test_scicrunch_service_api.py
+"""
+
+import logging
+from typing import Any, Dict, List, Union
+
+from aiohttp import ClientSession
+from pydantic import BaseModel, Field
+from yarl import URL
+
+from ._config import SciCrunchSettings
+from .models import ResourceHit
+
+logger = logging.getLogger(__name__)
+
+
+# MODELS --
+#
+# NOTE: These models are a trucated version of the data payload for a scicrunch response.#
+# NOTE: Examples of complete responses can be found in test_scicrunch.py::mock_scicrunch_service_api
+#
+
+
+class FieldItem(BaseModel):
+    field_name: str = Field(..., alias="field")
+    required: bool
+    value: Union[str, None, List[Any]] = None
+
+
+class ResourceView(BaseModel):
+    resource_fields: List[FieldItem] = Field([], alias="fields")
+    version: int
+    curation_status: str
+    last_curated_version: int
+    scicrunch_id: str
+
+    @classmethod
+    def from_response_payload(cls, payload: Dict):
+        assert payload["success"] == True  # nosec
+        return cls(**payload["data"])
+
+    @property
+    def is_curated(self) -> bool:
+        return self.curation_status.lower() == "curated"
+
+    def _get_field(self, fieldname: str):
+        for field in self.resource_fields:
+            if field.field_name == fieldname:
+                return field.value
+        raise ValueError(f"Cannot file expected field {fieldname}")
+
+    def get_name(self):
+        return str(self._get_field("Resource Name"))
+
+    def get_description(self):
+        return str(self._get_field("Description"))
+
+    def get_resource_url(self):
+        return URL(str(self._get_field("Resource URL")))
+
+
+class ListOfResourceHits(BaseModel):
+    __root__: List[ResourceHit]
+
+
+# REQUESTS
+
+
+async def get_all_versions(
+    unprefixed_rrid: str, client: ClientSession, settings: SciCrunchSettings
+) -> List[Dict[str, Any]]:
+    async with client.get(
+        f"{settings.SCICRUNCH_API_BASE_URL}/resource/versions/all/{unprefixed_rrid}",
+        params={"key": settings.SCICRUNCH_API_KEY.get_secret_value()},
+        raise_for_status=True,
+    ) as resp:
+        body = await resp.json()
```

Comment:
Please note that response can be plaintext. Did you consider handling this case?
The same applies for all the cases below.

Reply:
it is a restful API and the openapi specs defines always a json response. If there is a text, the error is captured above as an unexpected response